[ROCm] Update script install_pip_packages.sh for Python 2.7

deven-amd opened the pull request:
Currently ROCm TF cotnainers use the `install_pip_packages.sh` script to install the Python pip package-manager.
* https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/Dockerfile.rocm#L102
* https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/install/install_pip_packages.sh#L20-L23

Starting 01/23/2021, we started getting the following error while building TF containers for ROCM CI (when calling `get-pip.py` for Python2)

```
Traceback (most recent call last):
  File "get-pip.py", line 24226, in <module>
    main()
  File "get-pip.py", line 199, in main
    bootstrap(tmpdir=tmpdir)
  File "get-pip.py", line 82, in bootstrap
    from pip._internal.cli.main import main as pip_entry_point
  File "/tmp/tmpWkL0gn/pip.zip/pip/_internal/cli/main.py", line 60
    sys.stderr.write(f"ERROR: {exc}")
```

The cause seems to be an update to the `get-pip.py` script, which now picks the version `pip-21.0` (previously it was `pip-20.3.4`).
`pip-21.0` drops support for Python2.7 (as indicated by the following warning message)
```
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020.
Please upgrade your Python as Python 2.7 is no longer maintained.
pip 21.0 will drop support for Python 2.7 in January 2021.
More details about Python 2 support in pip can be found at
https://pip.pypa.io/en/latest/development/release-process/#python-2-support
pip 21.0 will remove support for this functionality.
```

It seems that there is now a Python 2.7 specific version of the `get-pip.py` script that we need to use, and that is what this commit does

deven-amd committed:
[ROCm] Update script install_pip_packages.sh for Python 2.7

Currently ROCm TF cotnainers use the `install_pip_packages.sh` script to install the Python pip package-manager.
* https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/Dockerfile.rocm#L102
* https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/install/install_pip_packages.sh#L20-L23

Starting 01/23/2021, we started getting the following error while building TF containers for ROCM CI (when calling `get-pip.py` for Python2)

```
Traceback (most recent call last):
  File "get-pip.py", line 24226, in <module>
    main()
  File "get-pip.py", line 199, in main
    bootstrap(tmpdir=tmpdir)
  File "get-pip.py", line 82, in bootstrap
    from pip._internal.cli.main import main as pip_entry_point
  File "/tmp/tmpWkL0gn/pip.zip/pip/_internal/cli/main.py", line 60
    sys.stderr.write(f"ERROR: {exc}")
```

The cause seems to be an update to the `get-pip.py` script, which now picks the version `pip-21.0` (previously it was `pip-20.3.4`).
`pip-21.0` drops support for Python2.7 (as indicated by the following warning message)
```
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020.
Please upgrade your Python as Python 2.7 is no longer maintained.
pip 21.0 will drop support for Python 2.7 in January 2021.
More details about Python 2 support in pip can be found at
https://pip.pypa.io/en/latest/development/release-process/#python-2-support
pip 21.0 will remove support for this functionality.
```

It seems that there is now a Python 2.7 specific version of the `get-pip.py` script that we need to use, and that is what this commit does

deven-amd commented:
Sample failure logs
* http://ml-ci.amd.com:21096/job/develop-upstream-unit-tests/job/tensorflow-upstream-unit-tests-cpu/134/
* http://ml-ci.amd.com:21096/job/develop-upstream-unit-tests/job/tensorflow-upstream-unit-tests-gpu-multi/126/ 
* http://ml-ci.amd.com:21096/job/develop-upstream-unit-tests/job/tensorflow-upstream-unit-tests-gpu-single/123/ 
* http://ml-ci.amd.com:21096/job/develop-upstream-sanity-tests/job/tensorflow-upstream-ubuntu-sanity-tests/68/

sunway513 commented:
@deven-amd , can you check if upstream TF still supports Python2.7?
It has been EOL for a while.

deven-amd committed:
Update script rocm_ci_sanity.sh to explicitly use Python3.5 version of the get-pip.py script

Starting 01/23/2021, we started getting the following error while building TF containers for ROCM CI (when calling `get-pip.py` within the `rocm_Ci_sanity.,sh` script)

```
Traceback (most recent call last):
  File "get-pip.py", line 24226, in <module>
    main()
  File "get-pip.py", line 199, in main
    bootstrap(tmpdir=tmpdir)
  File "get-pip.py", line 82, in bootstrap
    from pip._internal.cli.main import main as pip_entry_point
  File "/tmp/tmpWkL0gn/pip.zip/pip/_internal/cli/main.py", line 60
    sys.stderr.write(f"ERROR: {exc}")
```

It seems that there is now a Python 3.5 specific version of the `get-pip.py` script that we need to use, and that is what this commit does

deven-amd commented:
@sunway513 TF dropped support for Python2.x, starting with `TensorFlow 2.2`

Think we need to re-visit the upstream scripts we are using to build the ROCm TF container....we may not be using the correct ones.

For the time being, the change in this PR will get the CI going again, and should serve as a temporary solution.

deven-amd commented:
TF 2.2.0 release notes - https://github.com/tensorflow/tensorflow/releases/tag/v2.2.0
